### PR TITLE
Move mod/notifications to Module\Notification

### PR DESCRIPTION
**Pull request by nupplaphil**
Part of #4090 

This move helps me as well to see it more clearly that there's a general difference between introduction notifies and all the other notifications :-)

=> tested local and on my friendica node (all tabs, including json and clicking around)

### CODE REVIEW

File: src/Module/BaseNotifications.php [hunk 1]
```diff
@@ -0,0 +1,165 @@
...
+	/** @var int The determined start item of the current page */
+	protected static $firstItemNum;
+
+	/**
+	 * Collects all notifies from the backend
+	 *
+	 * @return array The determined notification array
+	 *               ['header', 'notifs']
+	 */
+	abstract public static function getNotifies();
```
Comment (MrPetovan) [inline src/Module/BaseNotifications.php]:
```suggestion
	abstract public static function getNotifications();
```

File: src/Module/BaseNotifications.php [hunk 2]
```diff
@@ -0,0 +1,165 @@
...
+	/** @var int The default count of items per page */
+	const ITEMS_PER_PAGE = 20;
+
+	/** @var boolean True, if ALL entries should get shown */
+	protected static $showAll;
+	/** @var int The determined start item of the current page */
+	protected static $firstItemNum;
+
+	/**
+	 * Collects all notifies from the backend
```
Comment (MrPetovan) [inline src/Module/BaseNotifications.php]:
```suggestion
	 * Collects all notifications from the backend
```

File: src/Module/BaseNotifications.php [hunk 3]
```diff
@@ -0,0 +1,165 @@
...
+	/** @var array Array of the allowed notifies and their printable name */
+	const PRINT_TYPES = [
+		Notify::NETWORK  => 'Network',
+		Notify::SYSTEM   => 'System',
+		Notify::HOME     => 'Home',
+		Notify::PERSONAL => 'Personal',
+		Notify::INTRO    => 'Introductions',
+	];
+
+	/** @var array The array of access keys for notify pages */
```
Comment (MrPetovan) [inline src/Module/BaseNotifications.php]:
```suggestion
	/** @var array The array of access keys for notification pages */
```

File: src/Module/BaseNotifications.php [hunk 4]
```diff
@@ -0,0 +1,165 @@
...
+	/** @var array Array of URL parameters */
+	const URL_TYPES = [
+		Notify::NETWORK  => 'network',
+		Notify::SYSTEM   => 'system',
+		Notify::HOME     => 'home',
+		Notify::PERSONAL => 'personal',
+		Notify::INTRO    => 'intros',
+	];
+
+	/** @var array Array of the allowed notifies and their printable name */
```
Comment (MrPetovan) [inline src/Module/BaseNotifications.php]:
```suggestion
	/** @var array Array of the allowed notifications and their printable name */
```

File: src/Module/BaseNotifications.php [hunk 5]
```diff
@@ -0,0 +1,165 @@
...
+
+namespace Friendica\Module;
+
+use Exception;
+use Friendica\BaseModule;
+use Friendica\Content\Pager;
+use Friendica\Core\Renderer;
+use Friendica\Core\System;
+use Friendica\DI;
+use Friendica\Model\Notify;
```
Comment (MrPetovan) [inline src/Module/BaseNotifications.php]:
```suggestion
use Friendica\Model\Notification;
```

File: src/Module/Notifications/Introductions.php [hunk 6]
```diff
@@ -0,0 +1,194 @@
...
+		// introduction
+		foreach ($notifs['notifications'] as $notif) {
+
+			// There are two kind of introduction. Contacts suggested by other contacts and normal connection requests.
+			// We have to distinguish between these two because they use different data.
+			switch ($notif['label']) {
+				case 'friend_suggestion':
+					$notif_content[] = Renderer::replaceMacros($sugg, [
+						'$type'           => $notif['label'],
+						'$str_notifytype' => DI::l10n()->t('Notification type:'),
```
Comment (MrPetovan) [inline src/Module/Notifications/Introductions.php]:
```suggestion
						'$str_notification_type' => DI::l10n()->t('Notification type:'),
```

File: src/Module/Notifications/Introductions.php [hunk 7]
```diff
@@ -0,0 +1,194 @@
...
+						'$notify_type'    => $notif['notify_type'],
```
Comment (MrPetovan) [inline src/Module/Notifications/Introductions.php]:
```suggestion
						'$str_type'    => $notif['str_type'],
```

File: src/Module/Notifications/Introductions.php [hunk 8]
```diff
@@ -0,0 +1,194 @@
...
+					if ($notif['network'] != Protocol::DIASPORA) {
+						$discard = DI::l10n()->t('Discard');
+					} else {
+						$discard = '';
+					}
+
+					$notif_content[] = Renderer::replaceMacros($tpl, [
+						'$type'           => $notif['label'],
+						'$header'         => $header,
+						'$str_notifytype' => DI::l10n()->t('Notification type:'),
```
Comment (MrPetovan) [inline src/Module/Notifications/Introductions.php]:
```suggestion
						'$str_notification_type' => DI::l10n()->t('Notification type:'),
```

File: src/Module/Notifications/Introductions.php [hunk 9]
```diff
@@ -0,0 +1,194 @@
...
+						'$notify_type'    => $notif['notify_type'],
```
Comment (MrPetovan) [inline src/Module/Notifications/Introductions.php]:
```suggestion
						'$str_type'    => $notif['str_type'],
```

File: src/Module/Notifications/Notifications.php [hunk 10]
```diff
@@ -0,0 +1,118 @@
...
+ * - Home
+ */
+class Notifications extends BaseNotifications
+{
+	/**
+	 * {@inheritDoc}
+	 */
+	public static function getNotifies()
+	{
+		$nm = DI::notify();
```
Comment (MrPetovan) [inline src/Module/Notifications/Notifications.php]:
```suggestion
		$nm = DI::notification();
```

File: src/Module/Notifications/Notifications.php [hunk 11]
```diff
@@ -0,0 +1,118 @@
...
+ * - System
+ * - Personal
+ * - Home
+ */
+class Notifications extends BaseNotifications
+{
+	/**
+	 * {@inheritDoc}
+	 */
+	public static function getNotifies()
```
Comment (MrPetovan) [inline src/Module/Notifications/Notifications.php]:
```suggestion
	public static function getNotifications()
```

File: src/Module/Notifications/Notifications.php [hunk 12]
```diff
@@ -0,0 +1,118 @@
...
+				$notification_templates = [
+					'like'        => 'notifications/likes_item.tpl',
+					'dislike'     => 'notifications/dislikes_item.tpl',
+					'attend'      => 'notifications/attend_item.tpl',
+					'attendno'    => 'notifications/attend_item.tpl',
+					'attendmaybe' => 'notifications/attend_item.tpl',
+					'friend'      => 'notifications/friends_item.tpl',
+					'comment'     => 'notifications/comments_item.tpl',
+					'post'        => 'notifications/posts_item.tpl',
+					'notify'      => 'notifications/notify.tpl',
```
Comment (MrPetovan) [inline src/Module/Notifications/Notifications.php]:
```suggestion
					'notification'      => 'notifications/notification.tpl',
```

File: static/routes.config.php [hunk 13]
```diff
@@ -167,6 +167,16 @@
...
+		'/system[/json]'     => [Module\Notifications\Notifications::class, [R::GET, R::POST]],
+		'/personal[/json]'   => [Module\Notifications\Notifications::class, [R::GET, R::POST]],
+		'/home[/json]'       => [Module\Notifications\Notifications::class, [R::GET, R::POST]],
+		'/intros[/json]'     => [Module\Notifications\Introductions::class, [R::GET, R::POST]],
+		'/intros/all[/json]' => [Module\Notifications\Introductions::class, [R::GET, R::POST]],
+		'/intros/{contact:\d+}[/json]' => [Module\Notifications\Introductions::class, [R::GET, R::POST]],
+	],
+
 	'/notify'         => [
 		'[/]'            => [Module\Notifications\Notify::class, [R::GET]],
```
Comment (MrPetovan) [inline static/routes.config.php]:
```suggestion
		'[/]'            => [Module\Notifications\Notification::class, [R::GET]],
```

File: view/theme/frio/templates/notifications/attend_item.tpl [hunk 14]
```diff
@@ -0,0 +1,2 @@
+
+{{include file="notifications/notify.tpl"}}
```
Comment (MrPetovan) [inline view/theme/frio/templates/notifications/attend_item.tpl]:
```suggestion
{{include file="notifications/notification.tpl"}}
```

File: view/theme/frio/templates/notifications/comments_item.tpl [hunk 15]
```diff
@@ -0,0 +1,2 @@
+
+{{include file="notifications/notify.tpl"}}
```
Comment (MrPetovan) [inline view/theme/frio/templates/notifications/comments_item.tpl]:
```suggestion
{{include file="notifications/notification.tpl"}}
```

File: view/theme/frio/templates/notifications/dislikes_item.tpl [hunk 16]
```diff
@@ -0,0 +1,2 @@
+
+{{include file="notifications/notify.tpl"}}
```
Comment (MrPetovan) [inline view/theme/frio/templates/notifications/dislikes_item.tpl]:
```suggestion
{{include file="notifications/notification.tpl"}}
```

File: view/theme/frio/templates/notifications/friends_item.tpl [hunk 17]
```diff
@@ -0,0 +1,2 @@
+
+{{include file="notifications/notify.tpl"}}
```
Comment (MrPetovan) [inline view/theme/frio/templates/notifications/friends_item.tpl]:
```suggestion
{{include file="notifications/notification.tpl"}}
```

File: view/theme/frio/templates/notifications/likes_item.tpl [hunk 18]
```diff
@@ -0,0 +1,2 @@
+
+{{include file="notifications/notify.tpl"}}
```
Comment (MrPetovan) [inline view/theme/frio/templates/notifications/likes_item.tpl]:
```suggestion
{{include file="notifications/notification.tpl"}}
```

File: view/theme/frio/templates/notifications/network_item.tpl [hunk 19]
```diff
@@ -0,0 +1,2 @@
+
+{{include file="notifications/notify.tpl"}}
```
Comment (MrPetovan) [inline view/theme/frio/templates/notifications/network_item.tpl]:
```suggestion
{{include file="notifications/notification.tpl"}}
```

File: view/theme/frio/templates/notifications/posts_item.tpl [hunk 20]
```diff
@@ -0,0 +1,2 @@
+
+{{include file="notifications/notify.tpl"}}
```
Comment (MrPetovan) [inline view/theme/frio/templates/notifications/posts_item.tpl]:
```suggestion
{{include file="notifications/notification.tpl"}}
```

File: view/theme/frio/templates/notifications/suggestions.tpl [hunk 21]
```diff
@@ -0,0 +1,2 @@
+
+{{include file="notifications/intros.tpl"}}
```
Comment (MrPetovan) [inline view/theme/frio/templates/notifications/suggestions.tpl]:
```suggestion
{{include file="notifications/notification.tpl"}}
```

Comment (nupplaphil) [reply view/theme/frio/templates/notifications/suggestions.tpl]: I think this change would be wrong, wouldn't it?

**Comment by nupplaphil** [comment]
Clicked around local and on my friendica.philipp.info node .. seems that I didn't brick something with the last changes :)

File: view/theme/frio/templates/notifications/suggestions.tpl (hunk 21 above)
Comment (MrPetovan) [reply view/theme/frio/templates/notifications/suggestions.tpl]: Huh, yeah, good point, copy and paste issue.

**Comment by nupplaphil** [comment]
So, I think I'm finished now..

The only occurrence is the "notify" table, but that's not something I do want to rename in this PR ..

**Comment by nupplaphil** [comment]
> Almost there!

you REALLY disliked the wrong use of `notify` and  `notifs`, didn't you ;-)

**Comment by MrPetovan** [comment]
Wait until I make you change all occurrences of "posting(s)" to just "post(s)".

**Comment by AlfredSK** [comment]
@nupplaphil Can you please have a look why the info bubble/badge on the notifications icon (the bell) doesn't work anymore? New notifications exist but the bubble doesn't show up. Thank you.

See https://forum.friendi.ca/display/0b6b25a8-135e-2c3c-46db-d09148468782

**Pull request merged**